components/canvas/Canvas.tsx

```diff
@@ -356,20 +366,38 @@ const Canvas = ({
         onNodeDragStop={onNodeDragStop}
         elevateEdgesOnSelect={true}
         edgesFocusable={userCanEdit}
-        onNodeMouseEnter={(_, node) => setHoveredNode(node)}
-        onNodeMouseLeave={() => setHoveredNode(undefined)}
-        onEdgeMouseEnter={(event, edge) => handleSetSelectedEdge(edge)}
+        onNodeMouseEnter={(_, node) => {
+          !menuData && setHoveredNode(node);
+        }}
+        onNodeMouseLeave={() => {
+          !menuData && setHoveredNode(undefined);
+        }}
+        onEdgeMouseEnter={(_, edge) => handleSetSelectedEdge(edge)}
         onEdgeMouseLeave={() => handleSetSelectedEdge(undefined)}
         attributionPosition="bottom-right"
         connectionRadius={100}
         nodeDragThreshold={15}
+        onNodeContextMenu={onNodeContextMenu}
+        onPaneContextMenu={onPaneContextMenu}
+        ref={ref}
       >
         <MiniMapCustom />
         <Controls className={styles.controls} showInteractive={false}>
           <ControlButton className={styles.zoomContainer}>
             <ZoomLevel />
           </ControlButton>
         </Controls>
+        {menuData && (
+          <ContextMenu
+            menuData={menuData}
+            copyToClipBoard={copyToClipboard}
+            paste={paste}
+            centerCanvas={centerCanvas}
```

Comment:
centerCanvas is fired upon initial render inside useCenterCanvas. This becomes a problem in ContetMenu since it renders every time the menu opens, causing the canvas to be centered. If you have a better suggestion of how to solve this, please tell.